src/components/DataTable/table-cells/TableHeadCell.tsx

```diff
@@ -29,11 +35,17 @@ export const TableHeadCell: React.FC<TableHeadCellProps> = ({
             className={clsx('moonstone-TableHeadCell', className)}
             iconStart={iconStart}
             iconEnd={iconEnd}
-            style={style}
             onClick={onClick}
         >
-            {children}
-            {renderSortIcon({sortDirection, isSorted})}
+            <span className="moonstone-TableHeadCell-content flexRow_nowrap alignCenter">
+                {children}
+                {SortIcon && (
+                    <SortIcon
+                        aria-label={`Icon for sorting in ${sortDirection} order`}
```

Comment:
- We should avoid hard-coding `label` as our product is internationalized, even if it was in the previous `SortIndicator` component.
- Instead we can add [aria-sort](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-sort) at the `th` level to define which column is sorted and its direction